src/app/Archives/AllTargetsArchivedRecordingsTable.tsx

```diff
@@ -170,18 +167,14 @@ export const AllTargetsArchivedRecordingsTable: React.FunctionComponent<AllTarge
 
     const handleLostTarget = React.useCallback(
       (target: Target) => {
-        let idx;
         setTargets((old) => {
-          for (idx = 0; idx < old.length; idx++) {
-            if (_.isEqual(target, old[idx])) break;
-          }
-          return old.filter((o) => !_.isEqual(o, target));
+          return old.filter((t) => !isEqualTarget(t, target));
         });
-        setExpandedTargets((old) => old.filter((o) => !_.isEqual(o, target)));
+        setExpandedTargets((old) => old.filter((t) => !isEqualTarget(t, target)));
         setCounts((old) => {
-          let updated = [...old];
-          updated.splice(idx, 1);
-          return updated;
+          const newMap = new Map<string, number>(old);
```

Comment:
I'm wondering if the map should be a `React.useState()` state instead of being generated every render. Is there something that prevents that from working?

Reply:
Oh, I see the state is a map, but you are creating a new one and using the old one as an argument.

Reply:
yeh I have to create new map since we have deps on hooks, which check if this map changes (i.e. obj ref).